NuRadioReco/modules/channelIceThermalNoiseAdder.py

```diff
@@ -0,0 +1,252 @@
+import functools
+import json
+import numpy as np
+import os
+import warnings
+
+import NuRadioReco.detector.antennapattern
+import NuRadioReco.framework.channel
+import NuRadioReco.framework.sim_station
+from NuRadioReco.modules.base.module import register_run
+from NuRadioReco.utilities import units, ice, geometryUtilities, signal_processing
+
+import logging
+logger = logging.getLogger('NuRadioReco.channelThermalNoiseAdder')
+
+
+class channelIceThermalNoiseAdder:
+    """
+    Module to generate thermal noise from ice
+    This class is a stripped version of channelGalacticNoiseAdder
+    Intead of using a skymap, the module uses pre generated effective antenna temperatures,
+    generated by Steffen's effective temperature code which was modified by Felix
+    (see NuRadioMC/examples/simulate_effective_ice_temperature )
+    """
+
+    def __init__(self):
+        self.__antenna_pattern_provider = NuRadioReco.detector.antennapattern.AntennaPatternProvider()
+
+
+    @functools.lru_cache(maxsize=1024 * 32)
+    def get_cached_antenna_response(self, antenna_pattern, zen, azi, *ant_orient):
+        return antenna_pattern.get_antenna_response_vectorized(self.freqs, zen, azi, *ant_orient)
+
+    def solid_angle(self, theta, d_theta, d_phi):
+        return np.abs(np.sin(theta) * np.sin(d_theta / 2) * 2 * d_phi)
```

Comment:
Maybe you also want to change this to:
```
return np.abs(np.sin(theta+d_theta / 2) * np.sin(d_theta / 2) * 2 * d_phi)
```
... otherwise you get a small shift (which will be negligible if you have very many theta bins)
(`np.sin(theta+d_theta / 2) * np.sin(d_theta / 2) * 2` is actually equivalent to `np.cos(theta)-np.cos(theta+d_theta)` ... personally I find the latter notation more intuitive)

Reply:
... or maybe you shouldn't since thetas are not calculated in the bin centers... but that was what could have also lead to Felix's confusion, since you always give the first "bin" a zero-temperature with the current implementation